application/engine/services/autopilot_daemon.py

```
@@ -1355,17 +1368,23 @@ async def _ensure_complete_ending(
+        truncated_by_token = stop_reason in ("length", "max_tokens", "MAX_TOKENS")
+
+        if not truncated_by_token:
+            if stop_reason:
+                # LLM 明确报告正常结束，不续写
+                return content
+            # stop_reason 为空（无法获取），降级到文本匹配
+            ending_pattern = r'[。！？…）】》"\'』」]$'
+            if re.search(ending_pattern, stripped):
+                return content
+            # 文本也不匹配结束符，可能是截断
+            logger.warning(f"[截断检测] stop_reason 未知且内容未以结束符结尾，尝试续写")
```

Ruff (0.15.12)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)